docs/concepts/internationalization.md

```diff
@@ -0,0 +1,72 @@
+Internationalization and localization
+=====================================
+
+Internationalization, i18n for short, is a way of designing code so that it requires no changes to be adapted for various languages. Localization is the process of adapting displayed text to the user's language.
+
+I18n is implemented using _translation keys_. A translation key is a string that identifies a piece of displayable text in no specific language. For example, `tile.dirt.name` is the language key referring to the name of the Dirt block. This way, displayable text may be referenced with no concern for a specific language. The code requires no changes to be adapted in a new language.
+
+Localization will happen in the game's locale. In a Minecraft client the locale is specified by the language settings. On a dedicated server, the only supported locale is en_US.
+
+Language files
+--------------
+
+Language files are located by `assets/domain/lang/locale.lang` (e.g. the US English translation for `examplemod` would be `assets/examplemod/lang/en_us.lang`). Resource pack format 3 requires the locale name to be lowercased. The file format is simply lines of key-value pairs encoded in UTF-8. Lines starting with `#` are treated as comments.
```

Comment:
Two questions:
1. All supported locales are defined in `pack.mcmeta` of Minecraft default resourcepack. [Here](https://minecraft.gamepedia.com/Language#Available_languages) is a (probably incomplete) list of valid locale codes. Should it be mentioned here?
2. The `*.lang` format only uses `=` to separate keys and values, and by default, ignores any invalid lines (i.e. lines that do not contains `=`), in addition to ignoring `#` comments. Should it be mentioned here?